File: Backend/backend/src/main/java/com/example/backend/entity/TenantAiUsage.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

/**
 * Per-tenant, per-month, per-feature AI usage + estimated cost. One row per
 * (tenant, year_month, feature); incremented atomically on every LLM call.
 * Feeds the (future) super-admin AI Economics view — cost per tenant/feature.
 */
@Entity
@Table(name = "tenant_ai_usage",
        uniqueConstraints = @UniqueConstraint(name = "uq_tenant_ai_usage",
                columnNames = {"tenant_id", "year_month", "feature"}))
@Data
@NoArgsConstructor
public class TenantAiUsage {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(name = "year_month", nullable = false, length = 7)
    private String yearMonth;   // e.g. "2026-06"

    @Column(nullable = false, length = 40)
    private String feature;     // COPILOT | BRIEFING | DESCRIBE_ITEM | REVIEW_DIGEST | REVIEW_REPLY | OTHER

    @Column(name = "call_count", nullable = false)
    private long callCount;

    @Column(name = "tokens_used", nullable = false)
    private long tokensUsed;

    @Column(name = "estimated_cost_rand", nullable = false)
    private double estimatedCostRand;

    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: SuperAdmin/backend/SuperAdmin.API/Models/TenantAiUsage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperAdmin.API.Models;

// Read-only view of the shared `tenant_ai_usage` table — per (tenant, month, feature) AI metering
// (call count, tokens, estimated Rand cost) written by the Spring backend for margin protection.
[Table("tenant_ai_usage")]
public class TenantAiUsage
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("tenant_id")]
    public Guid TenantId { get; set; }

    [Column("year_month")]
    public string YearMonth { get; set; } = "";

    [Column("feature")]
    public string Feature { get; set; } = "";

    [Column("call_count")]
    public long CallCount { get; set; }

    [Column("tokens_used")]
    public long TokensUsed { get; set; }

    [Column("estimated_cost_rand")]
    public double EstimatedCostRand { get; set; }

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }
}
